EmptyForInitializerPad: ArrayIndexOutOfBoundsException

Identified at https://github.com/checkstyle/checkstyle/pull/12277 ,

````
$ cat TestClass.java
public class TestClass {
    void method() {
for (
;;) {}
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
         <module name="EmptyForInitializerPad">
         </module>
    </module>
</module>

$ java -jar checkstyle-10.3.4-all.jar -c TestConfig.xml TestClass.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing TestClass.java
    at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:307)
    at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:224)
    at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:416)
    at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:339)
    at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:196)
    at com.puppycrawl.tools.checkstyle.Main.main(Main.java:131)
Caused by: java.lang.ArrayIndexOutOfBoundsException: origin(0) > fence(-1)
    at java.base/java.util.Spliterators.checkFromToBounds(Spliterators.java:387)
    at java.base/java.util.Spliterators.spliterator(Spliterators.java:239)
    at java.base/java.util.Arrays.spliterator(Arrays.java:5521)
    at java.base/java.util.Arrays.stream(Arrays.java:5663)
    at com.puppycrawl.tools.checkstyle.utils.CodePointUtil.hasWhitespaceBefore(CodePointUtil.java:54)
    at com.puppycrawl.tools.checkstyle.checks.whitespace.EmptyForInitializerPadCheck.visitToken(EmptyForInitializerPadCheck.java:157)
    at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:336)
    at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:407)
    at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:274)
    at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:155)
    at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
    at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:335)
    at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:294)
    ... 5 more
Checkstyle ends with 1 errors.
````

I am expecting no violation.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForInitializerPadCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForInitializerPadCheck.java
@@ -152,7 +152,7 @@ public class EmptyForInitializerPadCheck
             final int[] line = getLineCodePoints(lineIdx);
             final int before = ast.getColumnNo() - 1;
             // don't check if semi at beginning of line
-            if (!CodePointUtil.hasWhitespaceBefore(before, line)) {
+            if (ast.getColumnNo() > 0 && !CodePointUtil.hasWhitespaceBefore(before, line)) {
                 if (option == PadOption.NOSPACE
                     && CommonUtil.isCodePointWhitespace(line, before)) {
                     log(ast, MSG_PRECEDED, SEMICOLON);

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptyforinitializerpad/InputEmptyForInitializerPadDefaultConfig.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptyforinitializerpad/InputEmptyForInitializerPadDefaultConfig.java
@@ -58,4 +58,9 @@ class InputEmptyForInitializerPadDefaultConfig
         ;; ) {
         }
     }
+
+    void method3() {
+for (
+;;) {}
+    }
 }

